projects/packages/stats/src/class-tracking-pixel.php

```diff
@@ -85,6 +87,41 @@ public static function build_view_data() {
 			}
 		}
 
+		if ( $is_archive ) {
+			if ( $wp_the_query->is_home() ) {
+				$view_data['home'] = '1';
+			} elseif ( $wp_the_query->is_archive() ) {
+				if ( $wp_the_query->is_date ) {
+					$query               = $wp_the_query->query;
+					$date_parts          = array_filter( array( $query['year'] ?? null, $query['monthnum'] ?? null, $query['day'] ?? null ) );
+					$date                = implode( '/', $date_parts );
+					$view_data['arch']   = 'date';
+					$view_data['arch_v'] = $date;
+				} elseif ( $wp_the_query->is_category ) {
+					$view_data['arch']   = 'cat';
+					$view_data['arch_v'] = $wp_the_query->query['category_name'];
+				} elseif ( $wp_the_query->is_tag ) {
+					$view_data['arch']   = 'tag';
+					$view_data['arch_v'] = $wp_the_query->query['tag'];
+				} elseif ( $wp_the_query->is_author ) {
+					$view_data['arch']   = 'author';
+					$view_data['arch_v'] = $wp_the_query->query['author_name'];
+				} elseif ( $wp_the_query->is_tax ) {
+					$query = $wp_the_query->query;
+					if ( is_array( $query ) && count( $query ) === 1 ) {
+						$view_data['arch']   = array_keys( $query )[0];
+						$view_data['arch_v'] = array_values( $query )[0];
+					}
+				}
+			} elseif ( $wp_the_query->is_404() ) {
+				$view_data['arch']   = 'err';
+				$view_data['arch_v'] = sanitize_text_field( wp_unslash( $_SERVER['REQUEST_URI'] ?? '' ) ); // send the url path
+			} elseif ( $wp_the_query->is_search() ) {
+				$view_data['arch']   = 'search';
+				$view_data['arch_v'] = sanitize_text_field( $wp_the_query->query['s'] );
+			}
+		}
+
 		return $view_data;
```

Comment:
Maybe a problem for another day, but I wonder if we should count the number of results the archive page got. Or at least have a boolean to indicate when there are zero results. That is not a 404, but feels like something the end user would like to know that there are users that are filtering on their site but not finding any results.

Reply:
That was easily added. You can review [my last commit](https://github.com/Automattic/jetpack/pull/42368/commits/3f6254d9179363c4f18b4ddcb2ee6eaffa04e4b3) that adds the number of posts in the results in the `arch_res` key. I am flagging the comment as resolved for now. let me know if there is anything regarding this that has to be adjusted.

Reply:
Looks great. Thanks.